src/components/monitoringTable/MonitoringTable.tsx

```diff
@@ -0,0 +1,185 @@
+  const { t } = useTranslation(['monitorTable', 'general'])
+  const [producersAutocompleteTextInput, setProducersAutocompleteTextInput] =
+    useAutocompleteTextInput()
+  const [updateSnackbar] = stores.useSnackbarStore((state) => [state.updateSnackbar])
+  const navigate = useNavigate()
+
+  async function fetchProducers(value: string): Promise<FilterOption[]> {
+    if (producersAutocompleteTextInput) {
+      let response = apiRequests.getProducers(value)
+      return response
+    }
+    return []
+  }
+
+  const { data: producerOptions } = useQuery({
+    queryKey: ['producers', producersAutocompleteTextInput],
+    queryFn: () => fetchProducers(producersAutocompleteTextInput),
+    onError: (error) => updateSnackbar(true, 'Errore nella richiesta', 'error'),
+  })
+
+  const headLabels = [
+    t('serviceName'),
+    t('serviceVersion'),
+    t('serviceProducer'),
+    t('serviceState'),
+    t('lastDetectionDate'),
+    '',
+  ]
+
+  const { filtersParams, ...handlers } = useFilters<EServiceListQueryFilters>([
+    {
+      name: 'eserviceName',
+      type: 'freetext',
+      label: t('serviceNameFilter'),
+    },
+    {
+      name: 'producerName',
+      type: 'autocomplete-single',
+      label: t('serviceProducerFilter'),
+      onTextInputChange: setProducersAutocompleteTextInput,
+      options: producerOptions ? producerOptions : [],
+    },
+    {
+      name: 'versionNumber',
+      type: 'numeric',
+      label: t('serviceVersionFilter'),
+    },
+    {
+      name: 'state',
+      type: 'autocomplete-multiple',
+      label: t('serviceStateFilter'),
+      options: [
+        { value: 'ONLINE', label: 'online' },
+        { value: 'OFFLINE', label: 'offline' },
+        { value: 'N_D', label: 'n/d' },
+      ],
+    },
+  ])
+
+  const { paginationParams, paginationProps, getTotalPageCount } = usePagination({
+    limit: import.meta.env.VITE_PAGINATION_LIMIT
+      ? parseInt(import.meta.env.VITE_PAGINATION_LIMIT)
+      : 10,
+  })
+
+  async function fetchServices(
+    paginationParams: any,
+    filtersParams: any
+  ): Promise<GetServicesResponseType> {
+    let payload = {
+      offset: paginationParams.offset,
+      limit: paginationParams.limit,
+      producerName: filtersParams.producerName,
+      versionNumber: filtersParams.versionNumber,
+      eserviceName: filtersParams.eserviceName,
+      state: filtersParams.state,
+    }
+    let response = apiRequests.getServicesApi(payload as getServicesType)
+    return response
+  }
+
+  const { data: services, isInitialLoading } = useQuery({
+    queryKey: ['services', paginationParams, filtersParams],
+    queryFn: () => fetchServices(paginationParams, filtersParams),
+    onError: (error) => updateSnackbar(true, 'Errore nella richiesta', 'error'),
+  })
+
+  const goDetails = (serviceId: string) =>
+    navigate(
+      resolvePathVariables('/monitoring/serviceDetails/:serviceId', {
+        serviceId: serviceId,
+      })
```

Comment:
`react-router-dom` espone già una funzione che fa la stessa cosa, [generatePath](https://v5.reactrouter.com/web/api/generatePath). In questo caso basterebbe anche una string interpolation, ma è a vostra discrezione. 

Technical debt.